### DISCUSSION TITLE
Unexpected entries in dokku user authorized_keys

### DISCUSSION THREAD

**Discussion by alancwoo** [dokku/dokku#4703] [Q&A] [answered]
I spun up a fresh Ubuntu 20.04.02 LTS server and deployed two apps (php and vue/static). Two days later, I noticed three mysterious public keys in `/home/dokku/.ssh/authorized_keys`. Are these keys normal or for some kind of dokku service or has my server been compromised?

Keys have been redacted although they’re just pub keys, can share if useful. The `jondo@debian` is particularly discomforting.

```

command="FINGERPRINT=SHA256:<redacted> NAME=\"admin1\" `cat /home/dokku/.sshcommand` $SSH_ORIGINAL_COMMAND",no-agent-forwarding,no-user-rc,no-X11-forwarding,no-port-forwarding ssh-rsa <redacted pub key>
command="FINGERPRINT=SHA256:<redacted> NAME=\"web-admin1\" `cat /home/dokku/.sshcommand` $SSH_ORIGINAL_COMMAND",no-agent-forwarding,no-user-rc,no-X11-forwarding,no-port-forwarding ssh-rsa <redacted pub key>
command="FINGERPRINT=SHA256:<redacted> NAME=\"web-admin2\" `cat /home/dokku/.sshcommand` $SSH_ORIGINAL_COMMAND",no-agent-forwarding,no-user-rc,no-X11-forwarding,no-port-forwarding ssh-rsa <redacted pub key> jondo@debian
```

The apps are simply a PHP CMS and a static Vue Frontend. They are using the letsencrypt and storage plugins.

I’m genuinely perplexed and quite concerned, and am hoping these are just related to dokku itself somehow but can’t seem to find any information regarding this.

**Comment by alancwoo** [comment]
Hello again, so clearly, as it turns out, this was the result of failing to complete the setup through the web installer - while this is indeed mentioned in the documentation, @josegonzalez it might be very useful to have a notice at completion of the cli install script execution to ensure that the user is doubly aware of this issue?

https://github.com/dokku/dokku/blob/f63c2d18a320fc7d358d03e40b83d629fb041109/bootstrap.sh#L115

```
echo "--> 🚨🚨🚨 WARNING: If you don't complete setup via the web installer your Dokku installation will remain vulnerable to anyone finding the setup page and inserting their key. Visit https://dokku.com/docs/getting-started/installation/#3-setup-ssh-key-and-virtualhost-settings for more details."
```

Or even a check when running dokku commands, a reminder that the web installer process is still running and exposed.

**Reply by josegonzalez** [reply]
Actually I'd like to remove it completely in favor of https://github.com/dokku/dokku/issues/2247

**Comment by josegonzalez** [comment, marked as answer]
The next release (0.25.0) removes the web installer completely - users will need to complete the install on the CLI - which hopefully resolves this issue for future users.

**Reply by alancwoo** [reply]
This is perfect, I'm in full agreement thank you :)
